site/src/sphinx/server-annotated-service.rst

```diff
@@ -748,6 +748,37 @@ order commented. It is also the same as the evaluation order of the converters.
                         new MyGlobalExceptionHandler5(),        // order 5
                         new MyGlobalExceptionHandler6());       // order 6
 
+Returning a response
+--------------------
+
+In the earlier examples, the annotated service methods only return :api:`HttpResponse`, however there are
+more types able to be returned from the annotated service.
+
+- :api:`HttpResponse` and :api:`AggregatedHttpMessage`
+
+  - It will be responded to the client without any modification. Exception handlers will be invoked
+    if an exception is raised while the response is being sent. If no message has been sent to the client yet,
+    the exception handler can send an :api:`HttpResponse` instead.
+
+- ``Publisher<?>``
+
+  - All objects which are produced by the publisher will be collected, then the collected ones will be
+    converted to an :api:`HttpResponse` by response converters. If a single object is produced, it will be
+    passed into the response converters as it is. But if multiple objects are produced, they will be passed
+    into the response converters as a list. If the producer produces an error, exception handlers will handle it.
+    Note that ``ObservableSource<?>`` will be treated in the same way as ``Publisher<?>`` if you use
+    ``armeria-rxjava``.
+
+- ``CompletionStage<?>``
+
+  - An object which is generated by the ``CompletionStage<?>`` will be converted to an :api:`HttpResponse`
+    by response converters. If the ``CompletionStage<?>`` fails, exception handlers will handle the cause.
+
+- Other types
+
+  - Other types will be converted to an :api:`HttpResponse` by response converters. If a service method
```

Comment:
How about `As described in Converting a Java object to an HTTP response, you can use any response types with response converters that convert them.`